gamification-github-services/src/main/java/org/exoplatform/gamification/github/services/impl/WebhookServiceImpl.java

```diff
@@ -0,0 +1,416 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.exoplatform.gamification.github.services.impl;
+
+import java.io.*;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+import io.meeds.gamification.utils.Utils;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.conn.HttpClientConnectionManager;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.DefaultConnectionReuseStrategy;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.protocol.HTTP;
+import org.exoplatform.commons.ObjectAlreadyExistsException;
+import org.exoplatform.commons.api.persistence.ExoTransactional;
+import org.exoplatform.commons.exception.ObjectNotFoundException;
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.gamification.github.exception.GithubConnectionException;
+import org.exoplatform.gamification.github.model.WebHook;
+import org.exoplatform.gamification.github.model.RemoteOrganization;
+import org.exoplatform.gamification.github.services.WebhookService;
+import org.exoplatform.gamification.github.storage.WebHookStorage;
+import org.exoplatform.social.core.service.LinkProvider;
+import org.json.JSONObject;
+
+import org.exoplatform.services.listener.ListenerService;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+
+import javax.ws.rs.core.MediaType;
+
+import static org.exoplatform.gamification.github.utils.Utils.*;
+
+public class WebhookServiceImpl implements WebhookService {
+
+  private static final Log      LOG                                = ExoLogger.getLogger(WebhookServiceImpl.class);
+
+  private static final String   GITHUB_API_URL                     = "https://api.github.com/orgs/";
+
+  private static final String[] GITHUB_EVENTS                      = { "push", "pull_request", "pull_request_review",
+      "pull_request_review_comment", "pull_request_review_comment" };
```

Comment:
please make this parameterable and could be injected by plugin. Example of plugin Parent class:

```java
public class GithubTriggerPlugin extends BaseComponentPlugin {

...
  public String getTriggerName() {
    return "pull_request";
  }
...
  public String parseGithubUserId(String payload) {
...
  }
...
}
```